### PR TITLE
627 create developer information endpoint

### PR DESCRIPTION
**Pull request by davydocsurg**
## Describe your changes
created developer information endpoint

## Issue ticket number and link
#627 

## How Has This Been Tested?
Tested with Postman

## Screenshots (if appropriate):
image

### CODE REVIEW

File: __tests__/developers.test.js [hunk 1]
```diff
@@ -1,16 +1,5 @@
...
-  getWords,
-} from './shared/commands';
-import {
-  developerData,
-  malformedDeveloperData,
-  wordId,
-  exampleId,
-} from './__mocks__/documentData';
+import { createDeveloper, getDeveloper, getExample, getExamples, getWord, getWords } from './shared/commands';
+import { developerData, malformedDeveloperData, wordId, exampleId } from './__mocks__/documentData';
```
Comment (ijemmao) [inline __tests__/developers.test.js:2]: revert these changes

File: firebase.json [hunk 2]
```diff
@@ -44,4 +44,4 @@
       "port": 8088
     }
   }
-}
+}
```
Comment (ijemmao) [inline firebase.json:47]: revert this change

File: src/middleware/validateApiKey.js [hunk 3]
```diff
@@ -64,28 +24,16 @@ export default async (req, res, next) => {
     if (!apiKey && isDevelopment) {
       apiKey = FALLBACK_API_KEY;
     }
-    if (!apiKey) {
-      throw new Error('X-API-Key Header doesn\'t exist');
-    }
```
Comment (ijemmao) [inline src/middleware/validateApiKey.js:69]: revert this change - we don't want to alter existing logic

I'm thinking this file shouldn't even be changed since there are changes to code used actively in production. feel free to reach out to me to get more clarification in you need some 🙇🏾‍♀️

**Review by ijemmao — CHANGES_REQUESTED** [review]

File: __tests__/developers.test.js [hunk 4]
```diff
@@ -19,13 +20,11 @@ describe('Developers', () => {
       expect(res.status).toEqual(200);
       expect(res.body.message).not.toEqual(undefined);
     });
-
     it('should throw an error while creating a new developer', async () => {
       const res = await createDeveloper(malformedDeveloperData);
       expect(res.status).toEqual(400);
       expect(res.body.error).not.toEqual(undefined);
     });
-
```
Comment (ijemmao) [inline __tests__/developers.test.js:28]: revert this line change

**Review by ijemmao — CHANGES_REQUESTED** [review]

File: __tests__/developers.test.js [hunk 5]
```diff
@@ -19,13 +20,11 @@ describe('Developers', () => {
       expect(res.status).toEqual(200);
       expect(res.body.message).not.toEqual(undefined);
     });
-
```
[the remark this answers is not in the document]
Comment (davydocsurg) [reply __tests__/developers.test.js:22]: @ijemmao, kindly explain. I don't understand this

File: firebase.json (hunk 2 above)
Comment (ijemmao) [reply firebase.json:47]: @davydocsurg please revert this change - there shouldn't be a diff on this line

File: src/middleware/validateApiKey.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ijemmao) [reply src/middleware/validateApiKey.js:69]: it doesn't exist in your project version because you deleted that line - but if you take a look at this file but on the main branch in GitHub, you will see that the following is still in this file:

```js
if (!apiKey) {
  throw new Error('X-API-Key Header doesn\'t exist');
}
```

you don't want to delete this line because we want to catch the case where no API token is provided and we throw an error. if we don't, then we could allow the application to throw an even more obscure and detrimental issue down the line

so my recommendation is that you make as little changes to this file as possible to avoid introducing new possible bugs

**Review by ijemmao — CHANGES_REQUESTED** [review]

File: src/middleware/validateApiKey.js [hunk 6]
```diff
@@ -64,28 +20,20 @@ export default async (req, res, next) => {
     if (!apiKey && isDevelopment) {
       apiKey = FALLBACK_API_KEY;
     }
+
     if (!apiKey) {
-      throw new Error('X-API-Key Header doesn\'t exist');
```
Comment (ijemmao) [inline src/middleware/validateApiKey.js:68]: this is going to fail linting - we don't use double quotes - you really don't need to make a change to this part of the code 😅

File: src/controllers/developers/utils.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  console.timeEnd('Finding developer account');
+  return developer;
+};
+
+export const checkDeveloperAPILimit = async (apiKey) => {
+  const developer = await findDeveloper(apiKey);
+
+  if (developer) {
+    if (developer.usage.count >= determineLimit(PROD_LIMIT)) {
+      throw new Error('You have exceeded your API limit. Please upgrade your plan.', { cause: 403 });
```
Comment (ijemmao) [inline src/controllers/developers/utils.js:54]: Let's not change error messages. This error message is misleading since developer currently don't have an option to upgrade their plan.

```suggestion
  throw new Error('You have exceeded your limit of requests for the day', { cause: 403 })
```

**Review by ijemmao — CHANGES_REQUESTED** [review]
super close!

File: firebase.json (hunk 2 above)
Comment (davydocsurg) [reply firebase.json:47]: Fixed, this should be Prettier's format changes

**Review by ijemmao — APPROVED** [review]
